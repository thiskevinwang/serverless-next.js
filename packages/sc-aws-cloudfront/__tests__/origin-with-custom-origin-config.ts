import {
  createComponent,
  assertHasCustomOriginConfig,
  assertCDWTHasCustomOriginConfig
} from "../test-utils";

import {
  mockCreateDistributionWithTags,
  mockUpdateDistribution,
  mockCreateDistributionWithTagsPromise,
  mockGetDistributionConfigPromise,
  mockUpdateDistributionPromise
} from "../__mocks__/aws-sdk.mock";

jest.mock("aws-sdk", () => require("../__mocks__/aws-sdk.mock"));

describe("Input origin with custom origin config", () => {
  let component;

  beforeEach(async () => {
    mockCreateDistributionWithTagsPromise.mockResolvedValueOnce({
      Distribution: {
        Id: "xyz"
      }
    });

    component = await createComponent();
  });

  it("creates distribution with custom origin config", async () => {
    await component.default({
      origins: [
        {
          url: "http://exampleorigin.com",
          protocolPolicy: "http-only"
        }
      ]
    });

    assertCDWTHasCustomOriginConfig(mockCreateDistributionWithTags, {
      OriginProtocolPolicy: "http-only"
    });

    expect(mockCreateDistributionWithTags.mock.calls[0][0]).toMatchSnapshot();
  });

  it("updates distribution", async () => {
    mockGetDistributionConfigPromise.mockResolvedValueOnce({
      ETag: "etag",
      DistributionConfig: {
        Origins: {
          Quantity: 0,
          Items: []
        }
      }
    });
    mockUpdateDistributionPromise.mockResolvedValueOnce({
      Distribution: {
        Id: "xyz"
      }
    });

    await component.default({
      origins: [
        {
          url: "https://exampleorigin.com",
          protocolPolicy: "http-only"
        }
      ]
    });

    await component.default({
      origins: [
        {
          url: "https://exampleorigin.com",
          protocolPolicy: "https-only"
        }
      ]
    });

    assertHasCustomOriginConfig(mockUpdateDistribution, {
      OriginProtocolPolicy: "https-only"
    });

    expect(mockUpdateDistribution.mock.calls[0][0]).toMatchSnapshot();
  });
});
