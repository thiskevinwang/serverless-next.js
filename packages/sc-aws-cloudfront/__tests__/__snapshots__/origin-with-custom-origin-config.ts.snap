// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input origin with custom origin config creates distribution with custom origin config 1`] = `
Object {
  "DistributionConfigWithTags": Object {
    "DistributionConfig": Object {
      "Aliases": Object {
        "Items": Array [],
        "Quantity": 0,
      },
      "CacheBehaviors": Object {
        "Items": Array [],
        "Quantity": 0,
      },
      "CallerReference": "1566599541192",
      "Comment": "",
      "CustomErrorResponses": Object {
        "Items": Array [],
        "Quantity": 0,
      },
      "DefaultCacheBehavior": Object {
        "AllowedMethods": Object {
          "CachedMethods": Object {
            "Items": Array [
              "HEAD",
              "GET",
            ],
            "Quantity": 2,
          },
          "Items": Array [
            "HEAD",
            "GET",
          ],
          "Quantity": 2,
        },
        "Compress": false,
        "DefaultTTL": 86400,
        "FieldLevelEncryptionId": "",
        "ForwardedValues": Object {
          "Cookies": Object {
            "Forward": "none",
          },
          "Headers": Object {
            "Items": Array [],
            "Quantity": 0,
          },
          "QueryString": false,
          "QueryStringCacheKeys": Object {
            "Items": Array [],
            "Quantity": 0,
          },
        },
        "LambdaFunctionAssociations": Object {
          "Items": Array [],
          "Quantity": 0,
        },
        "MaxTTL": 31536000,
        "MinTTL": 0,
        "RealtimeLogConfigArn": undefined,
        "ResponseHeadersPolicyId": "",
        "SmoothStreaming": false,
        "TargetOriginId": "exampleorigin.com",
        "TrustedSigners": Object {
          "Enabled": false,
          "Items": Array [],
          "Quantity": 0,
        },
        "ViewerProtocolPolicy": "redirect-to-https",
      },
      "Enabled": true,
      "HttpVersion": "http2",
      "Origins": Object {
        "Items": Array [
          Object {
            "CustomHeaders": Object {
              "Items": Array [],
              "Quantity": 0,
            },
            "CustomOriginConfig": Object {
              "HTTPPort": 80,
              "HTTPSPort": 443,
              "OriginKeepaliveTimeout": 5,
              "OriginProtocolPolicy": "http-only",
              "OriginReadTimeout": 30,
              "OriginSslProtocols": Object {
                "Items": Array [
                  "TLSv1.2",
                ],
                "Quantity": 1,
              },
            },
            "DomainName": "exampleorigin.com",
            "Id": "exampleorigin.com",
            "OriginPath": "",
          },
        ],
        "Quantity": 1,
      },
      "PriceClass": "PriceClass_All",
    },
    "Tags": Object {
      "Items": Array [],
    },
  },
}
`;

exports[`Input origin with custom origin config updates distribution 1`] = `
Object {
  "DistributionConfig": Object {
    "CacheBehaviors": Object {
      "Items": Array [],
      "Quantity": 0,
    },
    "Comment": "",
    "CustomErrorResponses": Object {
      "Items": Array [],
      "Quantity": 0,
    },
    "DefaultCacheBehavior": Object {
      "AllowedMethods": Object {
        "CachedMethods": Object {
          "Items": Array [
            "HEAD",
            "GET",
          ],
          "Quantity": 2,
        },
        "Items": Array [
          "HEAD",
          "GET",
        ],
        "Quantity": 2,
      },
      "Compress": false,
      "DefaultTTL": 86400,
      "FieldLevelEncryptionId": "",
      "ForwardedValues": Object {
        "Cookies": Object {
          "Forward": "none",
        },
        "Headers": Object {
          "Items": Array [],
          "Quantity": 0,
        },
        "QueryString": false,
        "QueryStringCacheKeys": Object {
          "Items": Array [],
          "Quantity": 0,
        },
      },
      "LambdaFunctionAssociations": Object {
        "Items": Array [],
        "Quantity": 0,
      },
      "MaxTTL": 31536000,
      "MinTTL": 0,
      "RealtimeLogConfigArn": undefined,
      "ResponseHeadersPolicyId": "",
      "SmoothStreaming": false,
      "TargetOriginId": "exampleorigin.com",
      "TrustedSigners": Object {
        "Enabled": false,
        "Items": Array [],
        "Quantity": 0,
      },
      "ViewerProtocolPolicy": "redirect-to-https",
    },
    "Enabled": true,
    "Origins": Object {
      "Items": Array [
        Object {
          "CustomHeaders": Object {
            "Items": Array [],
            "Quantity": 0,
          },
          "CustomOriginConfig": Object {
            "HTTPPort": 80,
            "HTTPSPort": 443,
            "OriginKeepaliveTimeout": 5,
            "OriginProtocolPolicy": "https-only",
            "OriginReadTimeout": 30,
            "OriginSslProtocols": Object {
              "Items": Array [
                "TLSv1.2",
              ],
              "Quantity": 1,
            },
          },
          "DomainName": "exampleorigin.com",
          "Id": "exampleorigin.com",
          "OriginPath": "",
        },
      ],
      "Quantity": 1,
    },
    "PriceClass": "PriceClass_All",
  },
  "Id": "xyz",
  "IfMatch": "etag",
}
`;
